### PR TITLE
Logging and SecretDetector revisions

### PR DESCRIPTION
**Pull request by SimbaGithub**

### CODE REVIEW

File: Snowflake.Data.Tests/SecretDetectorTest.cs [hunk 1]
```diff
@@ -87,51 +87,91 @@ public void TestAWSKeys()
 
             // aws_key_id with double quotes on key
             BasicMasking(@"""aws_key_id"":'aaaaaaaa'", @"""aws_key_id"":'****'");
+
+            //If attribute is enclose in simple or double quote
```
Comment (sfc-gh-hchaturvedi) [inline Snowflake.Data.Tests/SecretDetectorTest.cs:91]: nit: typo: `single`

File: Snowflake.Data.Tests/SecretDetectorTest.cs [hunk 2]
```diff
@@ -87,51 +87,91 @@ public void TestAWSKeys()
...
+            BasicMasking(@"'aws_key_id'='aaaaaaaa'", @"'aws_key_id'='****'");
+            BasicMasking(@"""aws_key_id""='aaaaaaaa'", @"""aws_key_id""='****'");
```
Comment (sfc-gh-hchaturvedi) [inline Snowflake.Data.Tests/SecretDetectorTest.cs:93]: so the key is in double quotes but the value is in single quotes?

**Review by sfc-gh-hchaturvedi — APPROVED** [review]
Minor comments .. Rest LGTM.

**Review by sfc-gh-hyu — APPROVED** [review]

File: Snowflake.Data.Tests/SecretDetectorTest.cs (hunk 2 above)
Comment (SimbaGithub) [reply Snowflake.Data.Tests/SecretDetectorTest.cs:93]: The test is based on the pattern for AWS key :         private static readonly string AWS_KEY_PATTERN = @"(aws_key_id|aws_secret_key|access_key_id|secret_access_key)('|"")?(\s*[:=]\s*)'([^']+)'";

I think Louie found a case when the key was in " and that's why he added it. We might need to add " for the value. I'll see if I can find the log line which was related.

Comment (sfc-gh-hchaturvedi) [reply Snowflake.Data.Tests/SecretDetectorTest.cs:93]: I think this is related to the request I made while he was fixing the nodejs driver. It's fine I remember now. There was a leak scare a while ago where the AWS Secret KEY showed up in the logs. The request was in response to that.

**Pull request merged**